src/js/components/Select/SelectContainer.js

```diff
@@ -382,11 +382,15 @@ const SelectContainer = forwardRef(
                   // as an option Button kind property.
                   let child;
                   if (children)
-                    child = children(option, index, options, {
-                      active: optionActive,
-                      disabled: optionDisabled,
-                      selected: optionSelected,
-                    });
+                    child = (
+                      <OptionBox selected={optionSelected}>
```

Comment:
Should we also be passing `{...selectOptionsStyle}`?

Reply:
I don't think we should pass `{...selectOptionsStyle}` because this adds padding to each option and if the user is passing a custom child I don't think grommet should automatically add padding on it.

Reply:
Hmm... tricky to know what the theme might have in `theme.select.options.container`. Agreed on padding being a problem. Not so sure though if it has border or background. I think if we add `selected` to `Button`, this aspect isn't an issue.

Reply:
`theme.select.options.container` has the following properties set: 
```
options: {
        container: {
          align: 'start',
          pad: 'small',
        },
```
But as you said, if we are doing the selected styling in the Button then this won't matter

Reply:
I was thinking we just add `${props => props.selected && selectedStyle}` to Button. That would leave the select options container styling to always be supplied unless the caller passes a children render function, keeping the `plain` nature of that path.